File: java/2a.java
// this actually needs to be named a2.java, but I like having my files nice and neat
import java.io.*;
import java.util.*;

public class a2 {
  public static void main(String[] args) throws FileNotFoundException {
    long sum = 0;
    
    InputStream lines = new FileInputStream("C:/Users/saanv/Downloads/input.txt");
    Scanner input = new Scanner(lines);
    String[] ids = input.nextLine().strip().split(",");
    for (String id : ids) {
      long start = Long.parseLong(id.substring(0, id.indexOf("-")));
      long stop = Long.parseLong(id.substring(id.indexOf("-") + 1));
      for (Long i = start; i <= stop; i++) {
        if ((Long.toString(i)).substring(0, (Long.toString(i)).length() / 2).equals((Long.toString(i)).substring((Long.toString(i)).length() / 2))) {
          sum += i;
        }
      }
    }

    System.out.print(sum);
  }
}

File: python/2a.py
# so just make a list from the input and separate it

with open("C:/Users/saanv/Downloads/input.txt", "r") as line:
  ids = line.readline().strip().split(",")

sum = 0
dupes = []

for id in ids:
  start = int(id[:id.find("-")])
  stop = int(id[id.find("-") + 1:])
  for i in range(start, stop): # iterate through each number in the ranges
    if i not in dupes:
      for j in range(1, len(str(i))//2): # create the chunk size
        if i not in dupes:
          for k in range(1, len(str(i))//2): # iterate through each chunk
            if str(i)[k:k+j] == str(i)[k+j:k+j+j] and str(i)[k:k+j] != str(i)[k+j+j:k+j+j+j] and i not in dupes:
              dupes.append(i)
              sum += i

print(dupes)
#print(sum)
